### DISCUSSION TITLE
Question about writing boolean value

### DISCUSSION THREAD

**Discussion by jylilaht** [FreeOpcUa/opcua-asyncio#565] [Q&A]
value = await myvarnode.get_value()
print("Old Value: " + value)
newvalue = input("New Value: ")
dv = ua.DataValue(ua.Variant(type(value)(newvalue), myvarianttype))
await myvarnode.write_attribute(ua.AttributeIds.Value, dv)

VariantType.Boolean does not work if newvalue is 0 or 1 or True or False etc. How should I input boolean values for this code?

VariantType.Int32 is working just fine with same code.

I can read value just fine, (old value shows "True" or "False") just not write so connection is working. I do not get any visible errors either. It just does not change value on server.

What I am guessing is that I am trying to input value in wrong format, but cannot figure out how.

**Comment by oroulet** [comment]
input returns a stirng. you need to write a bool object. you can use a serach engine to learn how to convert a string to a boolean

**Reply by oroulet** [reply]
also there are 100s tests for boolean in test directoy. I am sure some of them use DataValue if you really need it

**Reply by oroulet** [reply]
ok looking at code over, maybe you convert, but your code is unnecessary complicated ad you give no info about whta error you get so it is hard and time consuming to help you

**Reply by jylilaht** [reply]
I'll try converting input to bool, could be a problem. 

My code tries to assign same varianttype that was read but input value is what it is. Weird is that int32 seems to work fine but maybe boolean don't have some automatic stuff.

I have to use code like this because simatic.net server timestamps had some sort of problem with default higher level functions so this is workaround.

**Comment by swamper123** [comment]
In python[ type()](https://docs.python.org/3/library/functions.html#type) doesn't cast a value. 
It isn't the same like in C (well Python is mostly C, but that is another topic 😉 ).
Like oroulet said, input() returns a String and you need a real boolean here.

**Reply by jylilaht** [reply]
yes this input conversion was my problem since this type() didn't work like I was thinking it would.

i just need to separate boolean and integer inputs somehow in code
